### PR TITLE
Doc: highlight rcparams

### PR DESCRIPTION
**Pull request by ImportanceOfBeingErnest**
## PR Summary

Many of the function parameters default to `None`, in which case the value from an rc parameter is taken. The custom `:rc:` role allows to link to the template, but it's always been bothering me that you then have to scroll through that complete file to locate the parameter you want to know the default of. 

This PR will highlight the parameter in question, such that it is easier to locate. 

I.e. if you click on a parameter, eg.

image

you will now be directed to 

    /tutorials/introductory/customizing.html?highlight=legend.numpoints#a-sample-matplotlibrc-file

and then scrolling down the file, the parameter in question is highlighted like this:

image

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
flake8 failure is real, but other than that I do like the idea a lot.

File: doc/_static/mpl.css [hunk 1]
```diff
@@ -594,6 +594,10 @@ span.linkdescr {
     font-size: 90%;
 }
 
+span.highlighted {
```
Comment (QuLogic) [inline doc/_static/mpl.css]: I don't understand why we need this; doesn't highlighting work already?

Comment (ImportanceOfBeingErnest) [reply doc/_static/mpl.css]: Yes and no. Somehow it does not seem to work (anymore?). For that reason it probably makes sense to only let it apply to those codelines in the rc params for now.

Comment (ImportanceOfBeingErnest) [reply doc/_static/mpl.css]: No actually, the highlighting was only for function signatures. So for example, https://matplotlib.org/1.4.1/api/axes_api.html?highlight=stem#matplotlib.axes.Axes.stem will show the stem line in yellow, but not every occurance of the word "stem". The signature highlighting probably got broken when intoducing the grey background for those. But I'm not sure how useful that is in general.

**Comment by ImportanceOfBeingErnest** [comment]
Actually, we could in addition directly include the default value in the rc role output:

image

What do you think?

File: doc/sphinxext/custom_roles.py [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 from os.path import sep
+from matplotlib import rcParamsDefault
 
 
 def rcparam_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
-    rendered = nodes.Text('rcParams["{}"]'.format(text))
+
+    param = rcParamsDefault.get(text)
+    if isinstance(param, str):
+        txt = f' : "{param}"'
```
Comment (anntzer) [inline doc/sphinxext/custom_roles.py]: 1) no space before the colon? (or use "=")
2) you can probably get away with `{param!r}` (aka `{repr(param)}`) which will give you the quotes for strings.

**Comment by jklymak** [comment]
The default may be a bit confusing?  Is it obvious this is marplotlibs default?  Not sure how to improve it though.  Maybe brackets?  Otherwise this seems great

**Comment by tacaswell** [comment]
Also, who ever merges this, please make sure I notice and re-build the 3.1.x docs.

**Comment by ImportanceOfBeingErnest** [comment]
Options:

* `rcParams["axes.facecolor"] = "white"`
* `rcParams["axes.facecolor"] : "white"`
* `rcParams["axes.facecolor"]: "white"`
* `rcParams["axes.facecolor"] ("white")`
* `axes.facecolor: "white"`
* `rcParams (axes.facecolor: "white")`

Personally I don't really care too much, so if anyone has a good argument for or against one or several of those...

**Comment by anntzer** [comment]
preference for "=" and ":" (without space before the colon), but anything is an improvement!

**Comment by ImportanceOfBeingErnest** [comment]
Ok, went for "=" now, because that actually gives valid python code:

image

**Comment by timhoffm** [comment]
This is great! :tada:.

> Ok, went for "=" now, because that actually gives valid python code:

If I was nitpicky, that is a bit misleading. Assignment is not what we want to express; and thinking of this, I have a tiny doubt if people would misunderstand the "=".

Anyway, "=" is what we are recommending in the [documentation guide](https://matplotlib.org/devdocs/devel/documenting_mpl.html#rcparams), so it's fine. But the documentation guide should be updated to reflect the new behavior.

**Comment by jklymak** [comment]
I would do something like (def.=xx) but maybe that’s too wordy

**Comment by timhoffm** [comment]
`(def.=xx)` is still too ambiguous. `def` is more commonly the abbreviation of definition rather than default; and this still has the slight equal sign issue.

If we want to be really clear `(default: xx)` would be the way to go. Thinking of it, I have a slight preference for that, but I'm fine to leave the current version as well.

**Comment by jklymak** [comment]
Dflt?

**Comment by timhoffm** [comment]
Hmmm, no. Readability counts also for the docs. We wouldn't use *dflt* as variable name. Why start saving letters in the documentation, which is allowed to be textually more verbose anyway.?

**Comment by tacaswell** [comment]
I like `=` as the line can now be copy-pasted to set the rcparam back to it's default.

**Comment by QuLogic** [comment]
Looking at the changes in #15230, the fact that `=` produces valid code seems to be a point _against_ using it. Many sentences appear to use constructs like:
```
Use :rc:`something` to do something.
```
or
```
You can do something by setting :rc:`something`.
```

These will become:
```
Use rcParams[something] = "default" to do something.
```
or
```
You can do something by setting rcParams[something] = "default".
```

This implies that you should set the rcParam to its default value, but those sentence often just mean set it to _anything_.
